05-JWT-Basics/starter/controllers/main.js

```diff
@@ -0,0 +1,98 @@
+
+// if exist create new JWT
+// send back to frontend
+// setup authentication so only the request with JWT can access the dasboard
+
+
+
+const jwt = require('jsonwebtoken') // import the jwt package
+// const CustomAPIError = require('../errors/custom-error')
+const {BadRequestError} = require('../errors')
+
+//check in the controllers
+
+const login = async(req,res)=>{
+
+    const { username, password } = req.body //// check username, password in post(login) request
+    console.log(username,password)
+
+    //check if username and password exist - if not throw error
+    if (!username || !password) {
+        // throw new CustomAPIError('Please provide email and password',400)
+        throw new BadRequestError('Please provide email and password')
+
+      }
+
+   
+   //just for demo for the payload, normally provided by DB!!!!
+    const id = new Date().getDate()
+
+  // try to keep payload small, better experience for user
+  // just for demo, in production use long secret, complex and unguessable string value!!!!!!!!!
+
+   //create new token
+  const token = jwt.sign(
+    { id, username }, //payload
+    process.env.JWT_SECRET, //jwt secret
+    { expiresIn: '30d',} //options
```

Comment:
I know the instructor hard-codes in the video, but this expired timeline can also be stored as an environment variable or declared as a constant

Reply:
Right - what's the importance of expired timeline  in environment variable? I did that as well on the preferred assignment :)